src/change_stream/event.rs

```diff
@@ -0,0 +1,119 @@
+//! Contains documents related to a ChangeStream event.
+use crate::coll::Namespace;
+use bson::{Bson, Document};
+use serde::{Deserialize, Serialize};
+
+/// An opaque token used for resuming an interrupted
+/// [`ChangeStream`](../struct.ChangeStream.html).
+///
+/// When starting a new change stream,
+/// [`start_after`](../option/struct.ChangeStreamOptions.html#structfield.start_after)
+/// and [`resume_after`](../option/struct.ChangeStreamOptions.html#structfield.resume_after) fields
+/// on [`ChangeStreamOptions`](../option/struct.ChangeStreamOptions.html) can be specified
+/// with instances of `ResumeToken`.
+///
+/// See the documentation
+/// [here](https://docs.mongodb.com/manual/changeStreams/#change-stream-resume-token) for more
+/// information on resume tokens.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct ResumeToken(pub(crate) Bson);
+
+/// A `ChangeStreamEvent` represents a
+/// [change event](https://docs.mongodb.com/manual/reference/change-events/) in the associated change stream.
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+#[non_exhaustive]
+pub struct ChangeStreamEvent<T> {
+    /// An opaque token for use when resuming an interrupted `ChangeStream`.
+    ///
+    /// See the documentation
+    /// [here](https://docs.mongodb.com/manual/changeStreams/#change-stream-resume-token) for
+    /// more information on resume tokens.
+    ///
+    /// Also see the documentation on [resuming a change
+    /// stream](https://docs.mongodb.com/manual/changeStreams/#resume-a-change-stream).
+    #[serde(rename = "_id")]
+    pub id: ResumeToken,
+
+    /// Describes the type of operation represented in this change notification.
+    pub operation_type: OperationType,
+
+    /// Identifies which collection or database where the event occurred.
+    pub ns: Option<ChangeStreamEventSource>,
+
+    /// The new name for the ns collection.  Only included for `OperationType::Rename`.
+    pub to: Option<Namespace>,
+
+    /// For unsharded collections this contains a single field, id, with the value of the id of the
+    /// document updated.  For sharded collections, this will contain all the components of the
+    /// shard key in order, followed by the id if the id isn’t part of the shard key.
+    pub document_key: Option<Document>,
```

Comment:
The doc for this is a little wordy; I like what the [manual](https://docs.mongodb.com/manual/reference/change-events/) has, can we replace it with something similar?

> A document that contains the `_id` of the document created or modified by the insert, replace, delete, update operations (i.e. CRUD operations). For sharded collections, also displays the full shard key for the document. The `_id` field is not repeated if it is already a part of the shard key.

Reply:
Done.